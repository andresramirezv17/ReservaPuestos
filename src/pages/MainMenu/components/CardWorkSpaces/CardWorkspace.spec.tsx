import React from 'react';
import { ReserveCardProps, CardWorkSpaces } from '.';
import { render, fireEvent } from '@testing-library/react';

describe('CardWorkspace user test', () => {
  let props: ReserveCardProps;
  beforeEach(() => {
    props = {
      reserve: {
        building: 'Torre Central',
        date: '10/09/2021',
        floor: 'Piso 1',
        section: 'SecciÃ³n A',
        workplace: 'P1A1',
        idReserve: 1,
        initialHour: '7:00',
        endHour: '15:00',
      },
      showModal: jest.fn(),
    };
  });

  it('it should match smnpashot', () => {
    const { container } = render(<CardWorkSpaces {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should call showModal when button is clicked', async () => {
    const { getByText } = render(<CardWorkSpaces {...props} />);
    const button = getByText('Realizar Check-in');
    fireEvent.click(button);
    expect(props.showModal).toHaveBeenCalled();
  });
});
