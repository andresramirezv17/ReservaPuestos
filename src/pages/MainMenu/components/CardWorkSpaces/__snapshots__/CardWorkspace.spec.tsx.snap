// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWorkspace user test it should match smnpashot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-mainBox-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-cardContent-2"
    >
      <div
        class="makeStyles-headerCard-3"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Edicio 
          Torre Central
        </h6>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Fecha: 
          10/09/2021
        </p>
      </div>
      <div
        class="makeStyles-bodyCard-4"
      >
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              style="font-weight: bold;"
            >
              Piso:
            </span>
             
            Piso 1
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              style="font-weight: bold;"
            >
              Puesto:
            </span>
             
            P1A1
          </p>
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              style="font-weight: bold;"
            >
              Sección: 
            </span>
            Sección A
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            <span
              style="font-weight: bold;"
            >
              Hora: 
            </span>
            7:00
             a 
            15:00
          </p>
        </div>
      </div>
      <div
        style="border-bottom: 1px solid grey; margin: 10px;"
      />
      <div
        class="makeStyles-footerCard-5"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonCheckin-7"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Realizar Check-in
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonChekout-8"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Finalizar Reserva
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
